src/main/java/build/buildfarm/common/config/Cas.java

```diff
@@ -16,12 +19,30 @@ public enum TYPE {
 
   private TYPE type = TYPE.FILESYSTEM;
   private String path = "cache";
-  private long maxSizeBytes = 2147483648L; // 2 * 1024 * 1024 * 1024
+  private long maxSizeBytes = 0;
   private long maxEntrySizeBytes = 2147483648L; // 2 * 1024 * 1024 * 1024
   private boolean fileDirectoriesIndexInMemory = false;
   private boolean skipLoad = false;
   private String target;
 
+  /*
+  Automatically set disk space to 90% of available space on the worker volume.
+  User configured value in .yaml will always take presedence.
+   */
+  public long getMaxSizeBytes() {
+    if (maxSizeBytes == 0) {
+      try {
+        maxSizeBytes =
+            (long)
+                (new File(BuildfarmConfigs.getInstance().getWorker().getRoot()).getFreeSpace()
```

Comment:
Seems like most people recommend `java.io.File.getUsableSpace()` more than instead of `java.io.File.getFreeSpace()` in https://stackoverflow.com/questions/21439596/difference-between-getfreespace-and-getusablespace-of-file

Reply:
Nice, good catch here!